changelog/unreleased/enhancement-app-menu-item-extension-point

```diff
@@ -0,0 +1,9 @@
+Enhancement: Application menu extension point
+
+We've added an extension point for adding items in the top left application switcher menu.
+
+`AppMenuItemExtension`'s can either link to an internal page (via `path`), an external page (via `url`) or have a button handler (via `handler`). Please refer to the docs for more information.
+
+DEPRECATION NOTICE: This deprecates the `applicationMenu` property of the appinfo object.
```

Comment:
I wanted to ask you to also add the deprecation to https://github.com/owncloud/web/issues/7338 but you already did that :D